pkg/connector/groups.go

```diff
@@ -106,25 +137,56 @@ func (o *groupBuilder) Grants(
 	annotations.Annotations,
 	error,
 ) {
-	bag := &pagination.Bag{}
-	err := bag.Unmarshal(pToken.Token)
-	if err != nil {
-		return nil, "", nil, err
-	}
-	if bag.Current() == nil {
-		bag.Push(pagination.PageState{
-			ResourceTypeID: groupResourceType.Id,
-		})
-	}
+	logger := ctxzap.Extract(ctx)
+	nextPage := ""
+	outputAnnotations := WithRateLimitAnnotations(nil)
+	var users []client.ConfluenceUser
 
-	users, token, ratelimitData, err := o.client.GetGroupMembers(
-		ctx,
-		bag.PageToken(),
-		resource.DisplayName,
-	)
-	outputAnnotations := WithRateLimitAnnotations(ratelimitData)
-	if err != nil {
-		return nil, "", outputAnnotations, err
+	// When we've detected groups containing forward slashes ('/') in the system,
+	// we cannot use the direct group members API endpoint for any group lookups
+	// due to a Confluence API limitation (see: https://jira.atlassian.com/browse/CONFCLOUD-68869).
+	// Instead, we use a cached approach that:
+	// 1. Lists all users in the system
+	// 2. For each user, fetches their group memberships
+	// 3. Builds a reverse mapping of group -> members
+	if !o.disableSlashSupportConfig && o.slashInGroupNameDetected {
+		logger.Debug("baton-confluence-datacenter: using cached group members list due to Confluence API limitation")
+		groupMembers, err := o.getGroupToMembersCache(ctx)
```

Comment:
The Grants() method on the user resource type can emit grants for other resources (such as groups). I think it would be more efficient to do that. See https://github.com/ConductorOne/baton-active-directory/blob/main/pkg/connector/users.go#L59 for an example of how that works.